### DISCUSSION TITLE
Get data across all processors in DAObjFunc

### DISCUSSION THREAD

**Discussion by uttamcadambi07** [mdolab/dafoam#414] [General]
Hi

Please refer to this line: [https://github.com/mdolab/dafoam/blob/b3f73e247456e7526789e0dc0458dc134c9474da/src/adjoint/DAObjFunc/DAObjFuncFieldInversion.C#L136](url)

Here, the field that is being referenced (UData) belongs to a processor when run in parallel. 
However, I would like to access the global UData field, meaning, everytime the else if statement is executed, I would like to gather all the data from all the processors and do some operations on the global UData field (I will need connectivity information because my operation requires the full field). How can this be accomplished? Is it as straightforward as doing reduce(UData, sumOp\<vector\>())?

**Comment by friedenhe** [comment]
We never gather field data to the master proc.. This should be an OpenFOAM question. I suggest you post it to cfd-online.com

**Reply by uttamcadambi07** [reply]
Thank you Dr. He. I have managed to find a solution to this!
